.github/workflows/test.yml

```diff
@@ -0,0 +1,54 @@
+name: CI
+
+on:
+  push:
+  pull_request:
+  # Allow manually triggering the workflow.
+  workflow_dispatch:
+
+# Cancels all previous workflow runs for the same branch that have not yet completed.
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+
+    strategy:
+      fail-fast: false
+      matrix:
+        php: ['7.3', '7.4', '8.0', '8.1']
+        composer-flags: ['']
+
+        include:
+          - php: '8.2'
+            composer-flags: '--ignore-platform-reqs'
+
+    name: PHP ${{ matrix.php }}
+
+    continue-on-error: ${{ matrix.php == '8.2' }}
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+
+      - name: Setup PHP
+        uses: shivammathur/setup-php@v2
+        with:
+          php-version: ${{ matrix.php }}
+          ini-values: zend.assertions=1, error_reporting=-1, display_errors=On
+          coverage: xdebug
+
+      - name: Validate Composer installation
+        run: composer validate --no-check-all --strict
```

Comment:
`--no-check-all` should be removed IMO. And `--no-check-lock` should be added, as the repo does not commit the lock file (and so there is no reason to validate it)

Reply:
Done

Reply:
FYI: `--no-check-lock` is redundant in a CI environment for repos without a committed `lock` file as it won't exist anyway.

Reply:
depends whether you run it before or after running a composer command that might generate it.
And it also makes the debugging clearer if someone tries to run the command locally.

Reply:
> depends whether you run it before or after running a composer command that might generate it.

True, but as its a separate job and that job doesn't run an `install`, that will never happen.

> And it also makes the debugging clearer if someone tries to run the command locally.

That's fair enough and a very valid point. 👍🏻 

Either way, the `--no-check-lock` is included in the workflow as it currently is.